server/upload/uploadserver/uploadserver.go

```diff
@@ -0,0 +1,121 @@
+package uploadserver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type Options struct {
+	UploadDir string
+}
+
+type Server struct {
+	Options
+	*mux.Router
+}
+
+func New(opts Options) *Server {
+	r := mux.NewRouter()
+	s := &Server{
+		Options: opts,
+		Router:  r,
+	}
+	r.Handle("/upload", handleError(s.handleUpload())).Methods("POST")
+	r.Handle("/", handleError(s.uploadForm())).Methods("GET")
+	r.Handle("/uploads/", http.StripPrefix("/uploads",
+		http.FileServer(http.Dir(s.Options.UploadDir))))
```

Comment:
You use `s.Options.UploadDir` here and `s.UploadDir` at L83. I prefer consistency in either case.